### PR TITLE
add chart for nydus-snapshotter

### PR DESCRIPTION
**Pull request by liubin**
Add new chart for nydus-snapshotter includes:

- daemonset for running nydus-snapshotter inside containers
- a configmap for configuring nydus-snapshotter

Fixes: #127 

Signed-off-by: bin liu <liubin0329@gmail.com>

### CODE REVIEW

File: charts/nydus-snapshotter/values.yaml [hunk 1]
```diff
@@ -0,0 +1,72 @@
+# nydus-snapshotter Helm Chart Values
+
+# -- Override nydus-snapshotter name
+nameOverride: ""
+# -- Override nydus-snapshotter fullname
+fullnameOverride: ""
+
+nydusSnapshotter:
```
Comment (gaius-qi) [inline charts/nydus-snapshotter/values.yaml:8]: `nydusSnapshotter` should be removed as there is only one nydusSnapshotter service, refer to https://github.com/bitnami/charts.

File: charts/nydus-snapshotter/values.yaml [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  # -- nydus-snapshotter name
+  name: nydus-snapshotter
```
Comment (gaius-qi) [inline charts/nydus-snapshotter/values.yaml:10]: `name` should be removed as there is only one nydusSnapshotter service, refer to https://github.com/bitnami/charts.

File: charts/nydus-snapshotter/values.yaml [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+  # -- Override nydus-snapshotter name
+  nameOverride: ""
```
Comment (gaius-qi) [inline charts/nydus-snapshotter/values.yaml:12]: `nameOverride` should be removed as there is only one nydusSnapshotter service, refer to https://github.com/bitnami/charts.

File: charts/nydus-snapshotter/values.yaml [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+  # -- Override nydus-snapshotter fullname
+  fullnameOverride: ""
```
Comment (gaius-qi) [inline charts/nydus-snapshotter/values.yaml:14]: `fullnameOverride` should be removed as there is only one nydusSnapshotter service, refer to https://github.com/bitnami/charts.

**Review by gaius-qi — CHANGES_REQUESTED** [review]

**Comment by nextauthn** [comment]
Nydus snapshotter doesn't share private registries creds with containerd.  Instead, this supports authentication in the following methods,

- Using $DOCKER_CONFIG or ~/.docker/config.json
- Proxying and scanning CRI Image Service API
- Using Kubernetes secrets (type = kubernetes.io/dockerconfigjson)

This chart has no place to set authentication. Moreover, is it possible to get secrets dynamically via kubelet client, as our dockerconfigjson is scheduled updated.

**Pull request merged**